File: 11-10-2024-reorganize-the-array.py
class Solution:
    def rearrange(self, ar):
        #Code here
        n = len(ar)
        for i in range(n):
            for j in range(n):
    
                # Check is any ar[j]
                # exists such that
                # ar[j] is equal to i
                if (ar[j] == i):
                    ar[j], ar[i] = ar[i], ar[j]
    
        # Iterate over array
        for i in range(n):
            
            # If not present
            if (ar[i] != i):
                ar[i] = -1
                
        return ar

File: 11-10-2024-reorganize-the-array.cpp
class Solution {
  public:
    vector<int> rearrange(vector<int>& arr) {
        // Code here
        int i, j, temp;
        int n = arr.size();
        // Iterate over the array
        for (i = 0; i < n; i++)
        {
            for (j = 0; j < n; j++)
            {
                // Check is any ar[j]
                // exists such that
                // ar[j] is equal to i
                if (arr[j] == i) {
                    temp = arr[j];
                    arr[j] = arr[i];
                    arr[i] = temp;
                    break;
                }
            }
        }
    
        // Iterate over array
        for (i = 0; i < n; i++)
        {
            // If not present
            if (arr[i] != i)
            {
                arr[i] = -1;
            }
        }
        
        return arr;
    }
};
